src/vcpkg/build.cpp

```diff
@@ -1096,6 +1095,14 @@ namespace vcpkg
         std::string portfile_cmake_contents;
         std::vector<Path> files;
         std::vector<std::string> hashes;
+
+        for (auto& file : abi_info.pre_build_info->hash_additional_files)
+        {
+            const auto hash =
+                vcpkg::Hash::get_file_hash(fs, file, Hash::Algorithm::Sha256).value_or_exit(VCPKG_LINE_INFO);
+            abi_tag_entries.emplace_back(Path(file).filename(), hash);
+        }
```

Comment:
```suggestion
        size_t i = 0;
        for (auto& file : abi_info.pre_build_info->hash_additional_files)
        {
            const auto hash =
                vcpkg::Hash::get_file_hash(fs, file, Hash::Algorithm::Sha256).value_or_exit(VCPKG_LINE_INFO);
            abi_tag_entries.emplace_back(fmt::format("additional_file_{}", i++), hash);
        }
```
Since we aren't including the paths to the files, we can't rely on the filenames being unique -- the user could have multiple files with the same name in different subdirectories.